smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy

```diff
@@ -191,7 +191,44 @@ map externalDocumentation {
 }
 
 /// Defines the ordered list of supported authentication schemes.
-@trait(selector: ":is(service, operation)")
+@trait(
+    selector: ":is(service, operation)"
+    breakingChanges: [
+        {
+            change: "add"
+            severity: "DANGER"
+            message: """
+                Adding the `@auth` trait will explicitly define the order of the \
+                authentication scheme traits for the target shape. \
```

Comment:
These messages seem to be geared towards the `service` shape. Seems like they'd need to be slightly different for the `operation` shape. You only have tests for the service shape cases. Maybe having operation shape cases would allow us to see if the error messages are helpful enough for the case.

Reply:
I also wonder if we should instead do this with the getEffectiveAuthSchemes approach to eliminate noise where there is no effective change. Like you have in the Migration classes, but generically, with sigv4 cases being special cases?

Reply:
I agree, the validation for `@auth` should be more comprehensive with `ServiceIndex::getEffectiveAuthSchemes` for both service and operation levels.

Reply:
Note: the `@auth` trait diff validation changes will be split into a different PR.

Reply:
See #2252